flink/v1.20/flink/src/test/java/org/apache/iceberg/flink/source/TestIcebergSourceSql.java

```diff
@@ -162,4 +173,74 @@ public void testWatermarkOptionsDescending() throws Exception {
         expected,
         SCHEMA_TS);
   }
+
+  @Test
+  public void testWatermarkInvalidConfig() {
+    CATALOG_EXTENSION.catalog().createTable(TestFixtures.TABLE_IDENTIFIER, SCHEMA_TS);
+
+    String flinkTable = "`default_catalog`.`default_database`.flink_table";
+    try {
+      SqlHelpers.sql(
+          getStreamingTableEnv(),
+          "CREATE TABLE %s "
+              + "(t1 TIMESTAMP(6), "
+              + "t2 BIGINT,"
+              + "eventTS AS CAST(t1 AS TIMESTAMP(3)), "
+              + "WATERMARK FOR eventTS AS SOURCE_WATERMARK()) WITH %s",
+          flinkTable,
+          SqlBase.toWithClause(getConnectorOptions()));
+
+      assertThatThrownBy(
+              () -> SqlHelpers.sql(getStreamingTableEnv(), "SELECT * FROM %s", flinkTable))
+          .isInstanceOf(NullPointerException.class)
+          .hasMessage("watermark-column needs to be configured to use source watermark.");
+    } finally {
+      SqlHelpers.sql(getStreamingTableEnv(), "DROP TABLE IF EXISTS %s", flinkTable);
```

Comment:
Why is this try finally? Can we just do a cleanup in an after method? That would make the tests easier to read

Reply:
Flink dynamic table is created only in specific tests in this class, so was handling here. But can do in after to improve readability.

Reply:
In `TestSqlBase` we have several tests where we create tables, and we expect that they are removed because of the `@TempDir` is removed. So we can simply just do the same here.

WDYT?